src/libs/zuora.ts

```diff
@@ -133,10 +133,19 @@ Additional Comms                                                   # reserved fo
     FROM
       RatePlanCharge
     WHERE
-      Product.ProductType__c = 'Newspaper - National Delivery' 
-      and RatePlanCharge.name = '${dayOfTheWeekName}' 
-      and RatePlanCharge.effectiveStartDate <= '${date}'
-      and (RatePlanCharge.effectiveEndDate >= '${date}' or (Subscription.autoRenew = true and Subscription.status = 'Active'))
+      (Subscription.Status = 'Active' OR Subscription.Status = 'Cancelled') AND
+      Product.Name = 'Newspaper - National Delivery' AND
+      RatePlanCharge.name = '${dayOfTheWeekName}' AND
+      RatePlanCharge.effectiveStartDate <= '${date}' AND
+      (
+          RatePlanCharge.effectiveEndDate > '${date}' OR
+          (
+              RatePlanCharge.EffectiveEndDate >= '${today}' AND
+              Subscription.Status = 'Active' AND
+              Subscription.AutoRenew = true AND
+              (RatePlan.AmendmentType IS NULL OR RatePlan.AmendmentType != 'RemoveProduct')
```

Comment:
if people switch products, the rateplans can be added or removed from the same subscription.  So the charge may be ending genuinely.